### PR TITLE
Rattlesnake retouch, Tamable Wildlife tamable rattlesnakes

### PR DESCRIPTION
**Pull request by randomtyper**
#### Summary
Mods "Tamable Wildlife: tamable rattlesnakes"

#### Purpose of change

Why tamable: small rattlesnakes have very low agression and the mod allows to tame much fiercer creatures.

Why retouch: rattlesnakes seem to not get much attention, their size jump from normal version is a bit too large (about 80x increase in body weight to 81 kg, yet HP is low for size at 30), and the evolved deathrattle lays cockatrice eggs for some reason. Also, the final evolution of rattler seems a bit underwhelming for a world where 500+ kg spiders are a thing (and 800 kg frogs).

#### Describe the solution

Non-mod related:
- [x] Rebalance rattlesnake weights&volumes;
- [x] New rattlesnakes (snakelets and mutant rattler);
- [x] Make all rattlers ovoviviparous (hatch eggs inside them);
- [x] Rebalance deathrattle to be the supersnake it should be.

Mod-related:
- [x] Rattlesnake pets!

#### Describe alternatives you've considered

Adding an ultra version of rattler (jormungandr?), a danger noodle weighing about 500-800 kg. Aborted, rebalanced current deathrattle to be the supersnake it should be.

Adding new type of eggs to deathrattle/making it lay rattler eggs. Aborted, all rattlers are ovoviviparous now (hatch eggs inside them). Can make it ovoparous as a result of mutation, though.

#### Testing

None. Will do when needed/possible.

#### Additional context

None.

### CODE REVIEW

**Comment by sadenar** [comment]
Nidhogg would probably be a better idea, but Jormundgandr imo would be better saved for a gigantic sea snake.
Sea snakes ranges do not extend to anywhere on the east coast (the very north of their American range is the south Californian coast) but it wouldn't be absurd for a nasty mutant snake line to be introduced to the waters.

**Comment by Karol1223** [comment]
Please for the love of god make normal descriptive names and not random mysticism bullshit. At the very list it's not a pun name but a bigass rattlesnake has nothing to do with mythological serpents. Especially not if we want to add more snakes in the future.

**Comment by randomtyper** [comment]
Aight, giant rattle snakelet (current rattlesnake) -> rattlesnake (current giant rattlesnake) -> giant rattlesnake (current mega rattlesnake/deathrattle) -> deathrattle (bumped up deathrattle). Agreed?

**Comment by Termineitor244** [comment]
Tamable snakes! Sounds good!

**Comment by Karol1223** [comment]
> Aight, giant rattlesnake snakelet (current rattlesnake) -> rattlesnake (current giant rattlesnake) -> giant rattlesnake (current mega rattlesnake/deathrattle) -> deathrattle (beefed up current deathrattle). Agreed?
> 
> Would mirror most creatures we have at this point (like spiderling -> spider -> giant spider -> mega spider).

Not quite. This should NOT mirror other mutants, because unlike with other mutants here you have a regular animal in the equation. And what you wrote isn't 100% accurate for spiders either.

What would be the best would be:
1. Normal rattlesnake, evolves into oversized rattlesnake if you want to take the long route, giant rattlesnake if you want it shorter. Lays rattlesnake eggs that hatch into rattlesnake hatchlings.
2. Oversized rattlesnake (if going the long route). At this point still would lay eggs that hatch into normal rattlesnakes I figure, since mutants start reproducing into mutants only at "Giant" (second size variation).
3. Giant rattlesnake. Should start laying eggs that hatch into its own mutant hatchlings.
4. Deathrattle. Should either lay more eggs that hatch into mutant hatchlings or be sterile (I know Venera said we wanted some largest mutants to be mutated beyond reproduction but idk)

**Comment by Karol1223** [comment]
Do me a solid while you're at it and name the "reptile" mutant eggs just mutant rattlesnake eggs. We do want other reptiles down the line. It'll get confusing as hell.

File: data/mods/Tamable_Wildlife/reptile_amphibian.json [hunk 1]
```diff
@@ -32,6 +32,63 @@
     "name": { "str": "colossal croaker" },
     "extend": { "flags": [ "CANPLAY", "PET_MOUNTABLE" ] }
   },
+  {
+    "id": "mon_rattlesnake",
+    "type": "MONSTER",
+    "copy-from": "mon_rattlesnake",
```
Comment (Karol1223) [inline data/mods/Tamable_Wildlife/reptile_amphibian.json:38]:
```suggestion
  {
    "id": "mon_rattlesnake_s",
    "type": "MONSTER",
    "copy-from": "mon_rattlesnake_s",
```

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/mods/Tamable_Wildlife/reptile_amphibian.json (hunk 1 above)
Comment (randomtyper) [reply data/mods/Tamable_Wildlife/reptile_amphibian.json]: Looked throught the file, thought I had a duped def, changed it manually.

**Comment by randomtyper** [comment]
> Do me a solid while you're at it and name (and ID) the "reptile" mutant eggs just mutant rattlesnake eggs. We do want other reptiles down the line. It'll get confusing as hell.
> 
> EDIT: Okay if they give live birth this is even more fun because now you should make our current reptile eggs not hatch! Surprise contribution motivation right here!

Filling up spawns for reptile eggs is now your problem tho. I am NOT doing that.

File: data/json/monstergroups/eggs.json [hunk 2]
```diff
@@ -88,7 +87,12 @@
   {
     "name": "GROUP_EGG_SNAKE",
     "type": "monstergroup",
-    "monsters": [ { "monster": "mon_rattlesnake" } ]
+    "monsters": [ { "monster": "mon_rattlesnake_s" } ]
+  },
+  {
+    "name": "GROUP_EGG_SNAKE_M",
+    "type": "monstergroup",
+    "monsters": [ { "monster": "mon_rattlesnake_big_s" } ]
```
Comment (Karol1223) [inline data/json/monstergroups/eggs.json:95]: I wanna say you can remove those entirely if you got rid of eggs spawning rattlesnakes. If you wanna be brave and get rid of the first one check in all of JSON if it's not used anywhere, though, please.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/monstergroups/eggs.json (hunk 2 above)
Comment (randomtyper) [reply data/json/monstergroups/eggs.json:95]: Yeah, I think I'll axe that. Those groups are not used anywhere anyway.

**Review by Karol1223 — CHANGES_REQUESTED** [review]
Okay so from my amateur point of view, most of this looks good. Personally, I'd also touch monstergroups to see if you're not missing out on the new size variation spawning somewhere in the world but other than that it looks fine. Also, does it really take a full year for Timber Rattlers to grow to adulthood? I'm no herpetologist, it may be the case, but it feels weirdly long.

EDIT: And I promptly forgot to include the actual review. Imo lower the dodge for biggun rattlers. They're big so they shouldn't be as dodge-y as the regular size one.

**Review by Karol1223 — APPROVED** [review]

**Pull request merged**